.github/workflows/check-react-native-nightly.yml

```diff
@@ -13,20 +13,27 @@ jobs:
     runs-on: macos-14
     strategy:
       matrix:
-        react-native-architecture: ["Paper", "Fabric"]
+        react-native-architecture: ['Paper', 'Fabric']
       fail-fast: false
     concurrency:
       group: ios-react-native-nightly-${{ matrix.react-native-architecture }}-${{ github.ref }}
       cancel-in-progress: true
     steps:
+      # Sometimes `npx react-native init` fails due to dependency mismatches or other
+      # rather vague errors. This is a workaround for that.
+      - name: Setup Yarn
+        run: yarn init
       - name: Create app
-        run: npx react-native init app --skip-install --version nightly
-      - name: Download Reanimated from commit
-        working-directory: app
-        run: curl -LJO https://codeload.github.com/software-mansion/react-native-reanimated/tar.gz/${{ github.sha }}
+        working-directory: ExpoApp
+        run: |
+          if npm view expo dist-tags | grep -q 'next:' ; then 
+            yarn install react-native@next && yarn react-native init app --skip-install --version next
+          else 
+            yarn install react-native@nightly && yarn react-native init app --skip-install --version nightly
```

Comment:
Maybe we should split it into two nightly CI check? 🤔

Reply:
I was thinking the same. 🤔

Reply:
so lets do it as a another PR 👍